packages/flutter/test/material/app_bar_test.dart

```diff
@@ -2563,6 +2563,223 @@ void main() {
+      final GlobalKey drawerListKey  = GlobalKey();
+      final GlobalKey bodyListKey = GlobalKey();
+       await tester.pumpWidget(MaterialApp(
+        home: Scaffold(
+          appBar: AppBar(
+            elevation: 0,
+            backgroundColor: MaterialStateColor.resolveWith((Set<MaterialState> states) {
+              return states.contains(MaterialState.scrolledUnder) ? scrolledColor : defaultColor;
+            }),
+            title: const Text('AppBar'),
+          ),
+          drawer: Drawer(
+            child: ListView(
+              key: drawerListKey,
+              children: <Widget>[
+                Container(height: 1200, color: Colors.red),
+              ],
+            ),
+          ),
+          body: ListView(
+            key: bodyListKey,
+            children: <Widget>[
+              Container(height: 1200, color: Colors.teal),
+            ],
+          ),
+        ),
+      ));
+
+      // Initial state: AppBar should have the default color.
+      expect(getAppBarBackgroundColor(tester), defaultColor);
+
+      // Scroll the list view.
+      await tester.drag(find.byKey(bodyListKey), const Offset(0, -300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should now have the scrolled color.
+      expect(getAppBarBackgroundColor(tester), scrolledColor);
+
+      // Open the drawer.
+      await tester.tap(find.byIcon(Icons.menu));
+      await tester.pumpAndSettle();
+
+      // The AppBar should still have the scrolled color.
+      expect(getAppBarBackgroundColor(tester), scrolledColor);
+
+      // Scroll the list inside the drawer.
+      await tester.drag(find.byKey(drawerListKey), const Offset(0, -300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should still have the scrolled color.
+      expect(getAppBarBackgroundColor(tester), scrolledColor);
+
+      // Scroll list inside the drawer back to the top.
+      await tester.drag(find.byKey(drawerListKey), const Offset(0, 300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should still have the scrolled color.
+      expect(getAppBarBackgroundColor(tester), scrolledColor);
+
+      // Close the drawer using the Scaffold's method.
+      tester.state<ScaffoldState>(find.byType(Scaffold)).closeDrawer();
+      await tester.pumpAndSettle();
+
+      // The AppBar should still have the scrolled color.
+      expect(getAppBarBackgroundColor(tester), scrolledColor);
+
+      // Scroll the list view back to the top.
+      await tester.drag(find.byKey(bodyListKey), const Offset(0, 300));
+      await tester.pumpAndSettle();
+
+      // The AppBar should be back to the default color.
+      expect(getAppBarBackgroundColor(tester), defaultColor);
+    });
+
+    testWidgets('Material3 - scrolledUnderElevation should be applied when scrolled under with multiple list views', (WidgetTester tester) async {
```

Comment:
```suggestion
    testWidgets('scrolledUnderElevation should be applied when scrolled under with multiple list views', (WidgetTester tester) async {
```

Material 3 is default now.